helm/cas-ciip-portal-data-migrate/templates/job.yaml

```diff
@@ -0,0 +1,156 @@
+                  key: credentials.json
+                  name: {{ .Values.existingGcpSecret }}
+            - name: WALG_GS_PREFIX
+              value: "gs://wksv3k-prod-ciip-backups/wal"
+            - name: WALE_GS_PREFIX
+              value: "gs://wksv3k-prod-ciip-backups/wal"
+          name: {{ template "cas-ciip-portal-data-migrate.fullname" . }}-prod-test-restore
+          image: {{ .Values.image.psql.repository }}:{{ .Values.image.psql.tag }}
+          resources:
+            limits:
+              cpu: 500m
+              memory: 1Gi
+            requests:
+              cpu: 100m
+              memory: 500Mi
+          command:
+            - /usr/bin/env
+            - bash
+            - -c
+            - |
+              set -euo pipefail;
+              if ! whoami &> /dev/null; then
+                if [ -w /etc/passwd ]; then
+                  echo "postgres:x:$(id -u):0:postgres user:/home/postgres:/sbin/nologin" >> /etc/passwd
+                fi
+              fi
+              rm -rf prod-bak
+              rm -rf credentials
+              echo ADDING GCS CREDENTIALS...
+              mkdir credentials
+              echo $GOOGLE_APPLICATION_CREDENTIALS_JSON > credentials/credentials.json
+              echo DOING WAL-G FETCH NOW...
+              wal-g backup-fetch prod-bak LATEST
+              cd prod-bak
+              echo ADDING recovery.signal...
+              touch recovery.signal
+              echo FIXING PG_HBA...
+              echo 'local all             all                                trust' > pg_hba.conf
+              cd ..
+              chmod -R 0750 ./prod-bak
+              echo STARTING SERVER IN RECOVERY MODE...
+              pg_ctl -D prod-bak start -w -o "-c archive_command='' -c ssl='off' -c hba_file='/home/postgres/prod-bak/pg_hba.conf' -c ident_file='/home/postgres/prod-bak/pg_ident.conf' --restore_command='wal-g wal-fetch %f %p' --recovery_target_action=promote"
+              echo RESTART POSTGRES IN NON-RECOVERY MODE...
+              pg_ctl -D prod-bak stop
+              cd prod-bak
+              rm recovery.signal
+              rm -rf backup_label
+              rm -rf backup_label.old
+              cd ..
+              pg_ctl -D prod-bak start -w -o "-c log_statement='all' -c logging_collector=on -c archive_mode='off' -c archive_command='' -c ssl='off' -c hba_file='/home/postgres/prod-bak/pg_hba.conf' -c ident_file='/home/postgres/prod-bak/pg_ident.conf'"
+
+              echo "$PORTAL_HOST:$PORTAL_PORT:$PORTAL_DATABASE:$PORTAL_USER:$PORTAL_PASSWORD" >> .pgpass
+              chmod 0600 .pgpass
+              # we need to recreate users first as the ACL created in sqitch need to be dumped to DB
+              # no need to grant any privileges on schemas or set passwords here, this will be done in the app-users job
+              PGPASSWORD="$PORTAL_PASSWORD" PGHOST=$PORTAL_HOST PGUSER=$PORTAL_USER psql<<EOF
+                create user $PORTAL_APP_USER;
```

Comment:
we have a tech debt card to remove this user as it is obsolete. Not sure if it should be removed here, or if for this migration job it doesn't matter.

Reply:
I don't think that this matters, unless we take care of the tech debt card before doing this migration